common/src/main/java/com/microsoft/identity/common/adal/internal/PowerManagerWrapper.java

```diff
@@ -130,8 +170,41 @@ public String getPowerOptimizationSettings(@NonNull final Context context){
      * @param connectionContext Context used to query if app is ignoring battery optimizations.
      * @return true if the given application package name is on the device's power allow list.
      */
-    @RequiresApi(Build.VERSION_CODES.M)
     public boolean isIgnoringBatteryOptimizations(final Context connectionContext) {
         return ((PowerManager) connectionContext.getSystemService(Context.POWER_SERVICE)).isIgnoringBatteryOptimizations(connectionContext.getPackageName());
     }
+
+    /**
+     * Checks if the app with the given package name is opted out from battery optimization.
+     * Caches the result in memory using computeIfAbsent for thread safety.
+     * Returns a string indicating the result or exception type.
+     *
+     * This is exposed to OneAuth.
+     *
+     * @param packageName The package name to check.
+     * @param context The context to use for PowerManager.
+     * @return a {@link BatteryOptimizationStatus}
+     */
+    public BatteryOptimizationStatus isAppOptedOutFromBatteryOptimization(@NonNull final String packageName, @NonNull final Context context) {
```

Comment:
>packageName [](http://example.com/codeflow?start=96&length=11)

PowerManager does not need the package name looking at the implementation. Are context and package for same app? can we just take context as input then?